sway_libs/src/merkle_proof/README.md

```diff
@@ -0,0 +1,92 @@
+# Overview
+
+Merkle trees allow for on-chain verification of off-chain data. With the merkle root posted on-chain, the generation of proofs off-chain can provide verifibly true data. 
+
+More information can be found in the [specification](./SPECIFICATION.md).
+
+## Known Issues
+
+Due to the Fuel VM padding of copy types smaller than one word, the current hashing of the leaf and node values do not match RFC-6962. The present implementation is the equavalent of hashing a `u64` as opposed to the standard `u8`. As a result, hashing of smaller sized copy types may result in unexpected behavior. For example, the `sha-256(0u8)` will result in the same hash as `sha-256(0u64)`. This will cause an incompatability with the [Fuel-Merkle](https://github.com/FuelLabs/fuel-merkle) repository. The issue regarding this can be tracked [here](https://github.com/FuelLabs/sway/issues/2594).
+
+# Using the Library
+
+## Using the Merkle Proof Library In Sway
+
+### Importing Into Your Project

```

Comment:
I'm unsure about this duplicated importing section. It's in the main readme and it's also in here. Perhaps the main readme should have a more generic example e.g. `use sway_libs::<library_name>::<library_function>`.
I think it's fine to have a generic explanation on the top level and then specific instructions in each library - if that is required. That being said, importing a library is the same everywhere so I'd consider removing it from here and having a generic example at the top level with perhaps 1 concrete example in case someone does not understand the generic syntax chosen by the author.